tests/RelayArgumentsOfCorrectType-test.ts

```diff
@@ -0,0 +1,113 @@
+import { RelayArgumentsOfCorrectType } from "../src/RelayArgumentsOfCorrectType"
+import { parse, buildSchema, validate } from "graphql"
+import { generateDirectivesFile } from "../src/generateDirectivesFile"
+import { readFileSync } from "fs"
+
+const schema = buildSchema(`
+${readFileSync(generateDirectivesFile(), "utf8")}
+
+type Foo {
+  bar: String
+  baz: Boolean
+}
+
+type Query {
+  foo: Foo
+  node(id: ID!): Foo
+  nodes(ids: [ID!]): [Foo]
+  conditionalNode(id: ID!, condition: Boolean!): Foo
+}
+`)
+
+function validateDocuments(source: string) {
+  return validate(schema, parse(source), [RelayArgumentsOfCorrectType])
+}
+
+describe(RelayArgumentsOfCorrectType, () => {
+  it("Disallows bad literal arguments", () => {
+    const errors = validateDocuments(`
+      fragment MyFragment on Query @argumentDefinitions(intVal: { type: "Int" }, reqStrVal: { type: "String!" }) {
+		  __typename
+	  }
+
+	  query MyQuery {
+		  ... MyFragment @arguments(intVal: "Test", reqStrVal: null)
+	  }
+    `)
+
+    expect(errors).toHaveLength(2)
+    expect(errors).toContainEqual(
+      expect.objectContaining({ message: 'Argument "intVal" for fragment "MyFragment" is expecting type "Int".' })
+    )
+    expect(errors).toContainEqual(
+      expect.objectContaining({
+        message: 'Argument "reqStrVal" for fragment "MyFragment" is expecting type "String!".',
+      })
+    )
+  })
+
+  it("Allows good literal arguments", () => {
+    const errors = validateDocuments(`
+      fragment MyFragment on Query @argumentDefinitions(intVal: { type: "Int" }, reqStrVal: { type: "String!" }) {
+		  __typename
+	  }
+
+	  query MyQuery {
+		  ... MyFragment @arguments(intVal: null, reqStrVal: "string")
+	  }
+    `)
+
+    expect(errors).toHaveLength(0)
+  })
+
+  // We should probably have a different validation rule for this. As far as I know this sort of stuff is not allowed within relay
```

Comment:
Good question. I do believe they relaxed these rules a bit recently, but not sure what they are off-hand.